contracts/utils/math/Math.sol

```diff
@@ -275,6 +283,24 @@ library Math {
         }
     }
 
+    /**
+     * @dev Returns the modular exponentiation of the specified base, exponent and modulus (b ** e % m)
+     *
+     * Requirements:
+     * - modulus can't be zero
+     * - result should be obtained successfully
+     */
+    function modExp(uint256 b, uint256 e, uint256 m) internal view returns (uint256) {
+        if (m == 0) {
+            revert MathModulusEqualsZero();
+        }
```

Comment:
The default behavior of the precompile is to return 0 when `m == 0`. Wouldn't make more sense to respect the precompile behavior and treat it as a noop? @Amxx

Reply:
Here, returning 0 if m == 0 would be like returning 0 when dividing something by 0.

Maybe at the precompile level its ok to do so, because they expect callers of the precompile will check they are not doing anything stupid, and because they don't want to handle reverts in precompile (just like ecrecover will return address(0) with invalid signatures).

But we are at the library level, and I think we should revert if someone tries to divide by 0 (or do modulo 0)

Reply:
Yeah but why is it stupid to allow `% 0`? If there's a concrete reason why this is dangerous, then let's keep it.

The thing is that I haven't come up with any good reason for either other than that it's difficult to remove the check once released 😅 

I agree with restricting the precompile at the library level, but the main difference with ECDSA is that it's dangerous to use it without the wrapper because there's a way of replaying a signature.

Reply:
> Yeah but why is it stupid to allow `% 0`?

`x % 0` is the remainder of the **division of x by 0**. What more can I say ?

Reply:
And? How is someone stupid for returning 0?

I am really sorry if this still doesn't make sense to you but I can't see a single reason why that is **dangerous** so that someone using it is stupid.

Reply:
Would you seriously consider that `mulDiv(1,1,0)` should return 0 as well?

Reply:
But 0 div panics by default. modexp doesn't.

As I said, I don't have reasons for saying it's okay or not. Doesn't seem harmful and restricting is irreversible.

Reply:
But 0 div panics by default. modexp doesn't.

- div is an opcode
- modexp is a precompile

Note that there is a mod opcode. If you try to use it to do `1 % 0`, this opcode will revert.

Reply:
interrestingly,

```
uint256 i = 1%0
``` 
does revert
```
unchecked { uint256 i = 1%0; }
```
does not

Reply:
Continuing the discussion on slack.